lib/vstd/DateUtils.cpp

```diff
@@ -1,33 +1,29 @@
 #include "StdInc.h"
 #include <vstd/DateUtils.h>
 
-#if defined(VCMI_ANDROID)
-#include "../CAndroidVMHelper.h"
-#endif
-
 VCMI_LIB_NAMESPACE_BEGIN
 
 namespace vstd
 {
 
-	DLL_LINKAGE std::string getFormattedDateTime(std::time_t dt)
+	DLL_LINKAGE std::string getFormattedDateTime(std::time_t dt, std::string format)
 	{
-#if defined(VCMI_ANDROID)
-		CAndroidVMHelper vmHelper;
-		return vmHelper.callStaticStringMethod(CAndroidVMHelper::NATIVE_METHODS_DEFAULT_CLASS, "getFormattedDateTime");
-#endif
-
 		std::tm tm = *std::localtime(&dt);
 		std::stringstream s;
-		try
-		{
-			s.imbue(std::locale(""));
-		}
-		catch(const std::runtime_error & e)
+		if(format.empty())
 		{
-			// locale not be available - keep default / global
+			try
+			{
+				s.imbue(std::locale(""));
+			}
+			catch(const std::runtime_error & e)
+			{
+				// locale not be available - keep default / global
+			}
```

Comment:
```suggestion
```
Can be removed since we no longer try to use system locale

Reply:
this code path is still used for `other_cp125*` languages

Reply:
other_* languages can't be selected here - those can only be selected for H3 encoding. Not for languages.

Reply:
Yes its an fallback. We could also remove and fallback to international format.